### PR TITLE
Make queries slimmer

### PR DESCRIPTION
**Pull request by joaquimrocha**
This PR decouples fetching the version breakdown + the instances stats from when getting a group's information.
Those queries were done in order to make it simpler for the UI to show the info, but they're a bit heavy so we should avoid them when they're not needed.

The PR also adds connections limits to the DB (overridable by the use of env vars).

### CODE REVIEW

File: frontend/src/js/components/Applications/ApplicationItemGroupItem.react.js [hunk 1]
```diff
@@ -12,15 +13,27 @@ const useStyles = makeStyles(theme => ({
 
 function ApplicationItemGroupItem(props) {
   const classes = useStyles();
-  const instances_total = props.group.instances_stats.total ? '(' + props.group.instances_stats.total + ')' : '';
+  const {group} = props;
+  const [totalInstances, setTotalInstances] = React.useState(-1);
```
Comment (ashu8912) [inline frontend/src/js/components/Applications/ApplicationItemGroupItem.react.js:17]: I think the default totalInstance count should be 0

File: frontend/src/js/components/Applications/ApplicationItemGroupItem.react.js [hunk 2]
```diff
@@ -12,15 +13,27 @@ const useStyles = makeStyles(theme => ({
...
+  [group]);
 
   return (
     <Link
       className={classes.groupLink}
       to={{pathname: `/apps/${props.group.application_id}/groups/${props.group.id}`}}
       component={RouterLink}
     >
-      {props.group.name} {instances_total}
+      {props.group.name} {(totalInstances > 0) && `(${totalInstances})`}
```
Comment (ashu8912) [inline frontend/src/js/components/Applications/ApplicationItemGroupItem.react.js:36]: I think template literals are not required here

File: frontend/src/js/components/Groups/Item.react.js [hunk 3]
```diff
@@ -21,13 +22,10 @@ const useStyles = makeStyles(theme => ({
 
 function Item(props) {
   const classes = useStyles();
+  const [totalInstances, setTotalInstances] = React.useState(-1);
```
Comment (ashu8912) [inline frontend/src/js/components/Groups/Item.react.js:26]: Same default should be 0

File: frontend/src/js/components/Instances/List.js [hunk 4]
```diff
@@ -95,6 +95,7 @@ function ListView(props) {
   const {application, group} = props;
   const [page, setPage] = React.useState(0);
   const [rowsPerPage, setRowsPerPage] = React.useState(10);
+  const [totalInstances, setTotalInstances] = React.useState(-1);
```
Comment (ashu8912) [inline frontend/src/js/components/Instances/List.js:99]: ^ default should be 0

File: frontend/src/js/components/Applications/ApplicationItemGroupItem.react.js (hunk 1 above)
Comment (joaquimrocha) [reply frontend/src/js/components/Applications/ApplicationItemGroupItem.react.js:17]: Nope. This is in order to give it an "unitialized" value (value yet not returned by the server), so we can decide to show something else based on that fact.

File: frontend/src/js/components/Applications/ApplicationItemGroupItem.react.js (hunk 2 above)
Comment (joaquimrocha) [reply frontend/src/js/components/Applications/ApplicationItemGroupItem.react.js:36]: What do you mean, I want the `(...)` to be shown only if the `totalInstances` are greater than 0.

File: frontend/src/js/components/Groups/Item.react.js (hunk 3 above)
Comment (joaquimrocha) [reply frontend/src/js/components/Groups/Item.react.js:26]: See my reply from before.

File: frontend/src/js/components/Instances/List.js (hunk 4 above)
Comment (joaquimrocha) [reply frontend/src/js/components/Instances/List.js:99]: See my reply from before.

File: frontend/src/js/components/Applications/ApplicationItemGroupItem.react.js (hunk 2 above)
Comment (ashu8912) [reply frontend/src/js/components/Applications/ApplicationItemGroupItem.react.js:36]: got it.

File: frontend/src/js/components/Applications/ApplicationItemGroupItem.react.js (hunk 1 above)
Comment (ashu8912) [reply frontend/src/js/components/Applications/ApplicationItemGroupItem.react.js:17]: hmm this is cool but than we need to decide what to show when there is a default state coz right now seeing -1 instances in UI looks very odd

Comment (joaquimrocha) [reply frontend/src/js/components/Applications/ApplicationItemGroupItem.react.js:17]: Oh! That's a different thing (and a good catch)! Let me check it out.

**Comment by joaquimrocha** [comment]
Merging after fixing the `-1` in the UI that @ashu8912 pointed out.

**Pull request merged**